pkg/executor/executor.go

```diff
@@ -1784,9 +1784,32 @@ func ResetContextOfStmt(ctx sessionctx.Context, s ast.StmtNode) (err error) {
 	}
 	sc.SetTimeZone(vars.Location())
 	sc.TaskID = stmtctx.AllocateTaskID()
-	sc.CTEStorageMap = map[int]*CTEStorages{}
+	if sc.CTEStorageMap == nil {
+		sc.CTEStorageMap = map[int]*CTEStorages{}
```

Comment:
```suggestion
		clear(sc.CTEStorageMap)
```

Reply:
`sc.CTEStorageMap` has a type of `any`.

Reply:
Can it defined as ```map[int]*executor.CTEStorages```

Reply:
I think so, but I'm not sure why it defined as `any` orignally. Let me check it later.

Reply:
I think it's defined as `any` because of the cycle dependency: `CTEStorages` is defined in `executor` and `executor` depends `stmtctx`. It's possible to make `sc.CTEStorageMap` as a concrete type by moving `CTEStorages` to `cteutil` package. However, I think it's unrelated to this PR, maybe we can take another PR to do it.